### PR TITLE
Adding recipe for reaper

### PR DESCRIPTION
**Pull request by jenzopr**
* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [X] This PR adds a new recipe.
* [X] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

This PR adds a recipe for the reaper, tally, minion and swan binaries from the kraken software tool set (http://www.ebi.ac.uk/research/enright/software/kraken). The tools are designed to streamline the analysis of next-generation sequencing data. Although designed with small RNA sequence analysis in mind the tools can be used to address issues facing next-generation sequencing in general.

Feedback is highly appreciated since this is my first PR to bioconda!
Best and thanks,
Jens

### CODE REVIEW

File: recipes/reaper/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+source:
+  url: http://wwwdev.ebi.ac.uk/enright-dev/kraken/reaper/src/reaper-15-065.tgz
+  md5: 33e1b4c8b72d6460d2fc3a45affaa8a2
+
+requirements:
+  build:
+    - gcc # [not osx]
+    - llvm # [osx]
+    - zlib
```
Comment (bgruening) [inline recipes/reaper/meta.yaml]: can you use `zlib 1.2.8`

File: recipes/reaper/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  run:
+    - libgcc # [not osx]
+    - zlib
```
Comment (bgruening) [inline recipes/reaper/meta.yaml]: here as well

File: recipes/reaper/build.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+cd src
+make
+mkdir -p $PREFIX/bin
+mv reaper $PREFIX/bin
+mv tally $PREFIX/bin
```
Comment (bgruening) [inline recipes/reaper/build.sh]: Should tally, swan and minion be in a separate repository?

**Comment by bgruening** [comment]
Thanks @jenzopr, a few comments inline.

File: recipes/reaper/build.sh (hunk 3 above)
Comment (jenzopr) [reply recipes/reaper/build.sh]: Thats a good point. They do not depend on each other, but can work together (especially reaper and tally, when it comes to trim paired-end sequencing files). The website states that `Tally and Minion are companion programs and shipped alongside reaper`.
How is this solved in other software tool sets? What would be the benefit to have them separately?

**Comment by jenzopr** [comment]
I wanted to continue the discussion that @bgruening started in the review above. The question is, if it makes more sense to pack reaper, tally and minion in seperate recipes or not. If not, we could enhance the visibility of the three programs by changing the name of to kraken (or ebi-kraken, since kraken is taken) to indicate that a tool set is being installed?

**Comment by bgruening** [comment]
@jenzopr you are the maintainer ... but I do not see a disadvantage in splitting this into 3 repos. The benefit, though only slightly, is that I only install what I need. I can upgrade versions independently, save some disc-space. Disadvantage for you is a larger maintenance overhead.
Not sure about visibility - will there be ever someone that searches for `reaper` and do not know about ebi-kraken?

**Comment by jenzopr** [comment]
Thanks a lot @bgruening! I agree that the benefit is a gain in flexibility. However, they are build together anyway and I guess that whoever looks for the toolset wants the full toolset. Independent upstream version upgrades will not very likely happen, since the tools are upgraded together (under the same version number).
I'll leave the name as is because reaper is the most prominent, and the source code fn has reaper inside and will add a note that tally, minion and swan come with the installation. How does that sound?

**Pull request closed without merging**